### ISSUE TITLE
Unable to initialize a constrained defaultdict

### ISSUE THREAD

**Issue by jeremyBanks** [Microsoft/pyright#22]
My code has a `collections.defaultdict` mapping from `str`s to `int`s, defaulting to `0` (`int()`) for missing values. I've been using the following annotation with MyPy to enforce those types.

```python
from collections import defaultdict
from typing import DefaultDict

x: DefaultDict[str, int] = defaultdict(int)
```

```none
$ mypy --strict example.py && echo "no errors"
no errors
```

However, Pyright tells me that the assignment is illegal.

```none
$ pyright example.py && echo "no errors"
Finding source files
Found 1 source files
Analyzed 1 file in 1.162sec
/Users/jeremybanks/example.py
  Expression of type 'defaultdict[Any, Any]' cannot be assigned to type 'DefaultDict[str, int]' (4, 4)
1 error, 0 warnings
```

**Comment by jeremyBanks** [comment]
That this may be a duplicate of #19, or it may requires further changes.

**Comment by erictraut** [comment]
I think this will be address by the fix I'm working on now. I'll confirm. Thanks for the bug report.

**Comment by erictraut** [comment]
This is fixed in 1.0.4, which I just published. Thanks for the bug report.
